src/main/java/io/coala/jetbrains/analysis/CodeAnalysisRunner.java

```diff
@@ -0,0 +1,147 @@
+package io.coala.jetbrains.analysis;
+
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.*;
+import com.intellij.openapi.components.ProjectComponent;
+import com.intellij.openapi.diagnostic.Logger;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.util.Key;
+import io.coala.jetbrains.settings.ProjectSettings;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+import java.util.concurrent.TimeUnit;
+
+public class CodeAnalysisRunner implements ProjectComponent {
+    private static final long timeOutInMilliseconds = TimeUnit.SECONDS.toMillis(120);
```

Comment:
Would be better if this was configurable. Add a `TODO` comment maybe or open a new issue.

Reply:
This could be added to the settings panel and then value taken from there.